### PR TITLE
Add BTRFS filesystem warning validation for VMware VMs

### PR DESCRIPTION
**Pull request by gwencasey96**
- Add validation policy to detect BTRFS filesystems in VMware guest disks
- Add comprehensive test suite with edge case coverage
- Warn users about potential migration issues with BTRFS
- Policy triggers conditional warnings only when BTRFS is detected

Generated with Cursor AI assistance

### CODE REVIEW

**Comment by yaacov** [comment]
@gwencasey96 hi, you need to sign your commits
`git commit -s`

the DCO error means a commit in this PR is not signed

**Comment by yaacov** [comment]
Can add some examples to the PR description above? for example:

this are the disks:
`oc mtv get inventory vms <provider> -o json | jq "< query for disks >" `
this are the alerts I got using this .rego file:
`oc mtv get inventory vms <provider> -o json | jq "< query for alerts >" `

File: validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+# Raise a concern for each BTRFS disk
+concerns[flag] {
+    btrfs_disks[idx]
+    disk := input.guestDisks[idx]
+    flag := {
+        "id": "vmware.guestDisks.btrfs.unsupported",
+        "category": "Warning",
+        "label": sprintf("BTRFS filesystem detected on disk '%v'", [disk.diskPath]),
```
Comment (mnecas) [inline validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego]: Think that showing the BTRFS is present in the VM should be enough. So we can rmeove the disk details

File: validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        "assessment": "BTRFS filesystem is not fully supported and may cause issues during migration. Consider converting to a supported filesystem like ext4 or XFS before migration."
```
Comment (mnecas) [inline validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego]: Please replace with: `BTRFS filesystem is not supported and may cause issues during guest conversion.`

File: validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego (hunk 1 above)
Comment (mnecas) [reply validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego]: @yaacov wdyt?

Comment (yaacov) [reply validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem.rego]: i am happy both ways, since we added the "id" field to collect same concerns together, having the strict label sting is not needed anymore and the label can be anything useful.

File: validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem_test.rego [hunk 3]
```diff
@@ -0,0 +1,235 @@
+package io.konveyor.forklift.vmware
+import future.keywords.in
+
+# Test: VM with no BTRFS disks should not trigger concerns
+test_vm_without_btrfs_no_concerns {
+    count(concerns) == 0 with input as {
```
[the remark this answers is not in the document]
Comment (yaacov) [reply validation/policies/io/konveyor/forklift/vmware/btrfs_filesystem_test.rego:6]: will be addressed by https://github.com/kubev2v/forklift/pull/2711

**Comment by yaacov** [comment]
@gwencasey96 hi,
a. please squash all commits, make sure the one commit left is signed
b. please rebase
c. please check failing tests

---

failing tests:
```
  validation/...lift/vmware/btrfs_filesystem_test.rego:69              | | Fail __local787__ = 2 with input as {"guestDisks": [{"capacity": 21474836480, "diskPath": "/dev/sda1", "filesystemType": "ext4", "key": 2000}, {"capacity": 42949672960, "diskPath": "/dev/sda2", "filesystemType": "btrfs", "key": 2001}, {"capacity": 10737418240, "diskPath": "/dev/sda3", "filesystemType": "xfs", "key": 2002}, {"capacity": 5368709120, "diskPath": "/dev/sda4", "filesystemType": "BTRFS", "key": 2003}], "guestId": "ubuntu64Guest"}
  query:1                                                              | Fail data.io.konveyor.forklift.vmware.test_mixed_filesystems_partial_concerns = _

data.io.konveyor.forklift.vmware.test_concern_structure_and_content#01: FAIL (5.748552ms)
```

**Review by mnecas — APPROVED** [review]

**Review by yaacov — CHANGES_REQUESTED** [review]
setting to request change so I will not merge by mistake before the tests are passing
@gwencasey96 see comment above with comments
